Scripts/assignment/emme_assignment.py

```diff
@@ -245,6 +254,65 @@ def _create_attributes(self, scenario, attributes):
                 scenario = scenario)
             log.debug("Created attr {} for scen {}".format(extr.name, scenario.id))
 
+    def calc_noise(self):
+        """Calculate noise according to Road Traffic Noise Nordic 1996.
+
+        Returns
+        -------
+        pandas.Series
+            Area (m2) of noise polluted zone, aggregated to area level
+        """
+        noise_areas = pandas.Series(0, zone_param.area_aggregation)
+        network = self.day_scenario.get_network()
+        morning_network = self.assignment_periods[0].emme_scenario.get_network()
+        for link in network.links():
+            # Aggregate traffic
+            light_modes = ("@car_work", "@car_leisure", "@van")
+            traffic = sum([link[mode] for mode in light_modes])
+            rlink = link.reverse_link
+            if rlink is None:
+                reverse_traffic = 0
+            else:
+                reverse_traffic = sum([rlink[mode] for mode in light_modes])
+            cross_traffic = (param.years_average_day_factor
+                             * param.share_7_22_of_day
+                             * (traffic+reverse_traffic))
+            heavy = link["@truck"] + link["@trailer_truck"]
+            traffic = max(traffic, 0.01)
+            heavy_share = heavy / (traffic+heavy)
+
+            # Calculate speed
+            link = morning_network.link(link.i_node, link.j_node)
+            rlink = link.reverse_link
+            if reverse_traffic > 0:
+                speed = 60 * 2 * link.length / (link.auto_time+rlink.auto_time)
+            else:
+                speed = 60 * (0.3*link.length/link.auto_time + 0.7*link.data2)
+            speed = max(speed, 50)
+
+            # Calculate start noise
+            if speed <= 90:
+                heavy_correction = (10*log10((1-heavy_share)
+                                    + 500*heavy_share/speed))
+            else:
+                heavy_correction = (10*log10((1-heavy_share)
+                                    + 5.6*heavy_share*(90/speed)**3))
+            start_noise = (68 + 30*log10(speed/50)
+                           + 10*log10(cross_traffic/15/1000)
+                           + heavy_correction)
+
+            # Calculate noise zone width
+            for i in param.noise_zone_width:
```

Comment:
(optional, highly personal preference) For me, `i` sounds like `integer` (but I guess some might interpret `i` as `item`, too). I'd prefer: `for noize_zone in param.noise_zone_width:`